autopush/endpoint.py

```diff
@@ -254,6 +271,41 @@ def _db_error_handling(self, d):
         d.addErrback(self._response_err)
         return d
 
+    def _process_auth(self, result):
```

Comment:
```python
if 'crypto-key' not in self.request.headers:
    return result
```

Would tidy this up quite a bit, the if header_info indented block wouldn't be needed, nor the crypto_key check at 286, nor the None assign to crypto_key. Adding a block to check for authorization and punt would remove that whole layer of indentation as well.

Yea, I know I'm a big fan of doing a check and returning early rather than adding levels of indentation. :)